spacy_curated_transformers/cli/fill_config_transformer.py

```diff
@@ -0,0 +1,469 @@
+    else:
+        msg.info(f"Looking up Hugging Face model name and revision in loader config...")
+        if loader_config is None or loader_config.get("name") is None:  # type: ignore
+            msg.fail(
+                "Pipeline config does not have a valid model loader configuration "
+                f"for the '{transformer_name}' Curated Transformer component. You can "
+                "provide the '--model-name' and '--model-revision' command-line "
+                "arguments to automatically fill in the loader configuration.",
+                exits=1,
+            )
+        assert loader_config is not None
+
+        model_name = loader_config["name"]
+        model_revision = loader_config.get("revision", "main")
+        msg.info(
+            title="Using loader Hugging Face model name and revision:",
+            text=f"{model_name} ({model_revision})",
+        )
+
+    return model_name, model_revision
+
+
+def _lookup_hf_model_type_for_curated_architecture(
+    msg: Printer, config: Config, transformer_name: str
+) -> str:
+    try:
+        transformer = config["components"][transformer_name]["model"]
+        trf_arch_splits = transformer["@architectures"].split(".")
+        assert len(trf_arch_splits) == 3 and trf_arch_splits[-2].endswith("Transformer")
+        curated_arch = trf_arch_splits[-2]
+    except (KeyError, AssertionError):
+        msg.fail(
+            "Pipeline config does not have a valid model architecture for "
+            f"the '{transformer_name}' Curated Transformer component",
+            exits=1,
+        )
+
+    hf_model_type = CURATED_TRANSFORMER_TO_HF_MODEL_TYPE.get(curated_arch)
+    if hf_model_type is None:
+        msg.fail(
+            "Missing Hugging Face model type for Curated Transformer "
+            f"model architecture '{curated_arch}'",
+            exits=1,
+        )
+    assert hf_model_type is not None
+    return hf_model_type
+
+
+def _validate_hf_model_type(
+    msg: Printer,
+    hf_config: Dict[str, Any],
+    expected_model_type: str,
+    transformer_name: str,
+):
+    incoming_hf_model_type = hf_config["model_type"]
+    if incoming_hf_model_type != expected_model_type:
+        error_msg = (
+            f"Hugging Face model of type '{incoming_hf_model_type}' cannot be loaded into "
+            f"Curated Transformer pipe '{transformer_name}' - "
+        )
+        hf_model_type_to_curated_arch = {
+            v: k for k, v in CURATED_TRANSFORMER_TO_HF_MODEL_TYPE.items()
+        }
+
+        if incoming_hf_model_type not in CURATED_TRANSFORMER_TO_HF_MODEL_TYPE.values():
+            error_msg += (
+                f"It is not supported by `spacy-curated-transformers`. The "
+                f"`{hf_model_type_to_curated_arch[expected_model_type]}` architecture "
+                f"expects a Hugging Face model of type '{expected_model_type}'"
+            )
+            msg.fail(
+                error_msg,
+                exits=1,
+            )
```

Comment:
I would structure this `if-else` and the error msg compilation slightly differently.
First, immediately after L288, check whether `incoming_hf_model_type` is supported, and if not, focus the error msg on that.

Otherwise, if it's valid/supported but the type is not the same as the `expected_model_type`, explain the incompatibility and recommend to either change the model name (according to the expected type) or change the architecture in the config.

Reply:
(this just feels easier to maintain to me, also avoiding the partly compiled error msg that is reused which makes it slightly more difficult to follow what exactly is being printed and when)

Reply:
Done.